lib/buffy/throttle_and_timed.ex

```diff
@@ -308,13 +348,14 @@ defmodule Buffy.ThrottleAndTimed do
 
       """
       @impl GenServer
-      @spec handle_cast(:throttle, Buffy.ThrottleAndTimed.state()) :: {:noreply, Buffy.ThrottleAndTimed.state()}
-      def handle_cast(:throttle, %{timer_ref: nil} = state) do
-        {:noreply, schedule_throttle_and_update_state(state)}
+      @spec handle_cast({:throttle, new_args :: any()}, Buffy.ThrottleAndTimed.state()) ::
+              {:noreply, Buffy.ThrottleAndTimed.state()}
+      def handle_cast({:throttle, new_args}, %{timer_ref: nil, args: args} = state) do
+        {:noreply, state |> schedule_throttle_and_update_state() |> Map.put(:args, update_args(args, new_args))}
       end
 
-      def handle_cast(:throttle, state) do
-        {:noreply, state}
+      def handle_cast({:throttle, new_args}, %{args: args} = state) do
+        {:noreply, %{state | args: update_args(args, new_args)}}
       end
```

Comment:
Something I'm thinking about is whether there might be issues with using `handle_cast` in a situation where were continuously getting throttle messages and the timeout occurs in the middle of a burst of them. Is there a chance we could drop some of the new things being added to state or overwrite them when we call `update_state_with_work_result/2`. 

I'd have to think about it and maybe test it out to know for sure if that's possible (or not possible), but maybe you've already considered this 🙂 and done they heavy mental lifting for me.

Reply:
I was trying to [test](https://github.com/stordco/buffy/blob/SIGNAL-5811-add-time-interval-bucket-feature-to-throttle-and-timed/test/buffy/throttle_and_timed_test.exs#L217) it and was wondering why I wasn't getting any new data into the state.

Then I realized that the beauty of `handle_continue` is that incoming messages will be kept getting stored in the inbox until `handle_continue` [finishes running](https://hexdocs.pm/elixir/1.12/GenServer.html#c:handle_continue/2). 

After the `handle_continue` is finished, the GenServer will then start ingesting the backed up messages in the inbox - this was confirmed by checking the `Process.info(pid, :message)`.

Reply:
> Then I realized that the beauty of handle_continue is that incoming messages will be kept getting stored in the inbox until handle_continue [finishes running](https://hexdocs.pm/elixir/1.12/GenServer.html#c:handle_continue/2).

❤️

Reply:
> After the handle_continue is finished, the GenServer will then start ingesting the backed up messages in the inbox - this was confirmed by checking the Process.info(pid, :message).

Could we measure/monitor the size of the inbox queue? That could be valuable lest we run into an issue where messages get dropped. I'm trying to find documentation on the behavior as the mailbox size grows but having trouble for some reason 🙃

Reply:
Thinking about it more, we could probably implement this in the handler itself during `handle_throttle` if we want to capture it